servers/rendering/renderer_rd/shaders/environment/gi.glsl

```diff
@@ -625,7 +626,11 @@ void process_gi(ivec2 pos, vec3 vertex, inout vec4 ambient_light, inout vec4 ref
 
 #ifdef USE_VOXEL_GI_INSTANCES
 		{
+#ifdef SAMPLE_VOXEL_GI_NEAREST
+			uvec2 voxel_gi_tex = texelFetch(usampler2D(voxel_gi_buffer, nearest_sampler), pos, 0).rg;
+#else
 			uvec2 voxel_gi_tex = texelFetch(usampler2D(voxel_gi_buffer, linear_sampler), pos, 0).rg;
```

Comment:
It's strange that this is causing validation errors. I was under the impression that `texelFetch` is supposed to ignore the sampler and the only reason you need to pass a sampler to `texelFetch` is that OpenGL only supported combined image samplers. 

Looking at the SPIR-V spec, `OpImageFetch` only requires an image of type `OpTypeImage` and not a sampled image of type `OpTypeSampledImage`.

Reply:
Looks like glslang generates really silly code for the `texelFetch` call:
```
         %13 = OpLoad %10 %voxel_gi_buffer
         %17 = OpLoad %14 %linear_sampler
         %19 = OpSampledImage %18 %13 %17
         %24 = OpImage %10 %19
         %26 = OpImageFetch %v4uint %24 %23 Lod %int_0
```
https://shader-playground.timjones.io/d63b2e6271a1ed699b18f59c3a995b01

Instead of passing the image directly to `OpImageFetch`, the code first creates a sampled image via `OpSampledImage` and then extracts the image from the sampled image via `OpImage` and then passes the result of `OpImage` to `OpImageFetch`

There appears to be a gslang extension that makes it possible to to use `texelFetch` without a sampler: [GL_EXT_samplerless_texture_functions](https://github.com/KhronosGroup/GLSL/blob/master/extensions/ext/GL_EXT_samplerless_texture_functions.txt)

Reply:
Very interesting.

I've updated the code to leverage that extension in the case where not linear filtering is supported. I'm kind of assuming it's always supported.